test/runner.bat

```diff
@@ -15,6 +18,27 @@
   set EM_PY=python
 )
 
+:: Work around Windows bug https://github.com/microsoft/terminal/issues/15212 : If this
+:: script is invoked via enclosing the invocation in quotes via PATH lookup, then %~f0 and
+:: %~dp0 expansions will not work.
+:: So first try if %~dp0 might work, and if not, manually look up this script from PATH.
+@if exist %~f0 (
+  set MYDIR=%~dp0
+  goto FOUND_MYDIR
+)
+for %%I in (%~n0.bat) do (
```

Comment:
What is this for loop looping over?

Reply:
It is looping over an array of length 1, that has the string `"%~n0.bat"` as its only entry.

This is because the construct `set MYDIR=%%~dp$PATH:I` can only be invoked for a variable `I` that has been created as an iterator variable in a `for` statement.